app/javascript/react/screens/App/Overview/screens/PlanWizard/PlanWizard.js

```diff
@@ -5,25 +5,109 @@ import { createMigrationPlans } from './helpers';
-  'planWizardScheduleStep',
-  'planWizardResultsStep'
-];
+import componentRegistry from '../../../../../../components/componentRegistry';
+import PlanWizardGeneralStep from '../PlanWizard/components/PlanWizardGeneralStep';
+import PlanWizardScheduleStep from '../PlanWizard/components/PlanWizardScheduleStep';
+
+const stepIDs = {
+  generalStep: 'planWizardGeneralStep',
+  vmStep: 'planWizardVMStep',
+  instancePropertiesStep: 'planWizardInstancePropertiesStep',
+  advancedOptionsStep: 'planWizardAdvancedOptionsStep',
+  scheduleStep: 'planWizardScheduleStep',
+  resultsStep: 'planWizardResultsStep'
+};
 
 class PlanWizard extends React.Component {
+  planWizardVMStepContainer = componentRegistry.markup('PlanWizardVMStepContainer');
+  planWizardResultsStepContainer = componentRegistry.markup('PlanWizardResultsStepContainer');
+  planWizardAdvancedOptionsStepContainer = componentRegistry.markup('PlanWizardAdvancedOptionsStepContainer');
+  planWizardInstancePropertiesStepContainer = componentRegistry.markup('PlanWizardInstancePropertiesStepContainer');
+
   state = { activeStepIndex: 0 };
 
+  getWizardSteps = () => {
+    const { planWizardGeneralStep, transformationMappings } = this.props;
+
+    const generalStep = {
+      id: stepIDs.generalStep,
+      title: __('General'),
+      render: () => <PlanWizardGeneralStep />,
+      disableGoto: !this.props.planWizardGeneralStep.values
+    };
+    const vmStep = {
+      id: stepIDs.vmStep,
+      title: __('VMs'),
+      render: () => this.planWizardVMStepContainer,
+      disableGoto: !this.props.planWizardVMStep.values
+    };
+    // eslint-disable-next-line no-unused-vars
+    const instancePropertiesStep = {
+      id: stepIDs.instancePropertiesStep,
+      title: __('Instance Properties'),
+      render: () => this.planWizardInstancePropertiesStepContainer,
+      disableGoto: true
+    };
+    const advancedOptionsStep = {
+      id: stepIDs.advancedOptionsStep,
+      title: __('Advanced Options'),
+      render: () => this.planWizardAdvancedOptionsStepContainer,
+      disableGoto: true
+    };
+    const scheduleStep = {
+      id: stepIDs.scheduleStep,
+      title: __('Schedule'),
+      render: () => <PlanWizardScheduleStep />,
+      disableGoto: true
+    };
+    const resultsStep = {
+      id: stepIDs.resultsStep,
+      title: __('Results'),
+      render: () => this.planWizardResultsStepContainer,
+      disableGoto: true
+    };
+
+    const selectedMappingId =
+      planWizardGeneralStep && planWizardGeneralStep.values && planWizardGeneralStep.values.infrastructure_mapping;
+    const selectedMapping = transformationMappings.find(mapping => mapping.id === selectedMappingId);
+
+    // eslint-disable-next-line no-unused-vars
+    const openstackTargetSelected =
+      selectedMapping &&
+      selectedMapping.transformation_mapping_items &&
+      selectedMapping.transformation_mapping_items.some(
+        item =>
+          item.destination_type === 'CloudTenant' ||
```

Comment:
Any chance these can be abstracted into the constants file?

Reply:
Good point. I'm not sure this is the final logic for this, though. I'll move them up anyway, can always change them.